nest-city-account/src/bloomreach/bloomreach.service.ts

```diff
@@ -166,14 +162,13 @@ export class BloomreachService {
     userId: string,
     isLegalPerson: boolean
   ) {
+    const userType = isLegalPerson ? 'legal person' : 'user'
     if (!cognitoId) {
-      if (isLegalPerson) {
-        this.logger.error(`Legal person ${userId} has no externalId, skipping trackEventConsents`)
-      } else {
-        this.logger.error(`User ${userId} has no externalId, skipping trackEventConsents`)
-      }
+      this.logger.error(
+        `No externalId for ${userType} with id: ${userId} has no externalId, skipping trackEventConsents`
+      )
     } else {
```

Comment:
I would prefer to remove this else, and have a return in the `if (!cognitoId) {` branch, so the readability and nesting can be improved.